### PR TITLE
CA-2039 Suppress cpu level logging if not available (or not rooted)

### PR DESCRIPTION
**Pull request by Tooseriuz**
This PR include with CA-2031, CA-2039, CA-2040 and CA-2053

### Changes
- Rename of role to the same as defined AppRole 
- Add function for check if rooted. will not getting cpu data if not rooted.
- Add permission checking before start the service
- set Bluetooth on when admin role is started

### CODE REVIEW

File: admin/src/main/java/org/rfcx/guardian/admin/RfcxGuardian.java [hunk 1]
```diff
@@ -86,6 +87,7 @@ public void onCreate() {
 		ShellCommands.triggerNeedForRootAccess(this);
 		DeviceI2cUtils.resetI2cPermissions(this);
 		DateTimeUtils.resetDateTimeReadWritePermissions(this);
+		DeviceBluetooth.setOn();
```
Comment (antonyharfield) [inline admin/src/main/java/org/rfcx/guardian/admin/RfcxGuardian.java]: There is a preference `admin_enable_bluetooth`. We should check that this is `true` before we call `setOn()`.

File: admin/src/main/java/org/rfcx/guardian/admin/activity/MainActivity.kt [hunk 2]
```diff
@@ -89,7 +89,9 @@ class MainActivity : AppCompatActivity() {
 
     public override fun onPause() {
         super.onPause()
-        monitorTimer.cancel()
+        if(::monitorTimer.isInitialized){
```
Comment (antonyharfield) [inline admin/src/main/java/org/rfcx/guardian/admin/activity/MainActivity.kt]: Please fix code style.

File: core/src/main/java/org/rfcx/guardian/utility/device/DeviceCPU.java [hunk 3]
```diff
@@ -69,29 +69,31 @@ private void incrementClockSpeedAverage() {
...
-	        reader.close();
-	        toks = load.split(" ");
-	        long idle2 = Long.parseLong(toks[5]);
-	        long cpu2 = Long.parseLong(toks[2]) + Long.parseLong(toks[3]) + Long.parseLong(toks[4])
-	            + Long.parseLong(toks[6]) + Long.parseLong(toks[7]) + Long.parseLong(toks[8]);
-	        return (float)(cpu2 - cpu1) / ((cpu2 + idle2) - (cpu1 + idle1));
-	    } catch (IOException e) {
-	    		RfcxLog.logExc(logTag, e);
-	    }
+		if (DeviceRoot.isRooted()) {
```
Comment (antonyharfield) [inline core/src/main/java/org/rfcx/guardian/utility/device/DeviceCPU.java]: Large blocks of code inside an `if` is a code smell.

You could rewrite this as:
```
if (! DeviceRoot.isRooted()) {
    return 0;
}
...
```

File: core/src/main/java/org/rfcx/guardian/utility/device/DeviceCPU.java [hunk 4]
```diff
@@ -69,29 +69,31 @@ private void incrementClockSpeedAverage() {
...
+				load = reader.readLine();
+				reader.close();
+				toks = load.split(" ");
+				long idle2 = Long.parseLong(toks[5]);
+				long cpu2 = Long.parseLong(toks[2]) + Long.parseLong(toks[3]) + Long.parseLong(toks[4])
+						+ Long.parseLong(toks[6]) + Long.parseLong(toks[7]) + Long.parseLong(toks[8]);
+				return (float) (cpu2 - cpu1) / ((cpu2 + idle2) - (cpu1 + idle1));
+			} catch (IOException e) {
+				RfcxLog.logExc(logTag, e);
+			}
```
Comment (antonyharfield) [inline core/src/main/java/org/rfcx/guardian/utility/device/DeviceCPU.java]: This code hasn't changed so it shouldn't be in git.

I think you have your Android Studio setup incorrectly @Tooseriuz -- you have converted 4 spaces into tabs. Please use 4 spaces to be consistent with the rest of the code.

File: core/src/main/java/org/rfcx/guardian/utility/device/DeviceRoot.java [hunk 5]
```diff
@@ -0,0 +1,19 @@
+package org.rfcx.guardian.utility.device;
+
+import android.os.Build;
+
+import java.io.File;
+
+public class DeviceRoot {
+    public static boolean isRooted() {
+        String buildTags = Build.TAGS;
+        if(buildTags != null && buildTags.contains("test-keys")) {
```
Comment (antonyharfield) [inline core/src/main/java/org/rfcx/guardian/utility/device/DeviceRoot.java]: Fix code style `if(`

File: core/src/main/java/org/rfcx/guardian/utility/device/DeviceRoot.java [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+            return true;
+        } else {
```
Comment (antonyharfield) [inline core/src/main/java/org/rfcx/guardian/utility/device/DeviceRoot.java]: The `else` is not required.

File: guardian/src/main/java/org/rfcx/guardian/guardian/utils/CheckAppPermissionUtils.kt [hunk 7]
```diff
@@ -0,0 +1,13 @@
+package org.rfcx.guardian.guardian.utils
+
+import android.content.Context
+import android.content.pm.PackageManager
+
+object CheckAppPermissionUtils {
+
+    fun checkLocationPermission(context: Context): Boolean{
```
Comment (antonyharfield) [inline guardian/src/main/java/org/rfcx/guardian/guardian/utils/CheckAppPermissionUtils.kt]: Code style `Boolean{`

**Review by antonyharfield — CHANGES_REQUESTED** [review]
Lots of good changes. :)

Important to fix the bluetooth setOn to only be when the preferences are set.

**Review by antonyharfield — CHANGES_REQUESTED** [review]
Let's remove the indent changes... it is better if the pull request is ONLY the changes related to the pull request.

**Review by antonyharfield — APPROVED** [review]
Ok - good to merge!

**Comment by Debjani1989** [comment]
Hello Rfcx/Rfcx-Guardian-Android team,

How are you doing? This is to inform you that I am not a part of this
operation anymore, but I am still a big fan and well wisher of rfcx and
Topher, no doubt. So would u please remove my email from the list? I am
getting continuous notification of the communication.

Thank you very much
Debjani Shyama
Social Media Associate
64-65 Kazi Nazrul Islam Avenue, Dhaka-1215, Bangladesh
Phone:01788 198233

**Comment by Tooseriuz** [comment]
Hi @Debjani1989,
You can unsubscribe this repository by choose unwatch here
image

**Pull request merged**